lms/djangoapps/certificates/views/webview.py

```diff
@@ -332,7 +333,11 @@ def _get_user_certificate(request, user, course_key, course, preview_mode=None):
     if preview_mode:
         # certificate is being previewed from studio
         if request.user.has_perm(PREVIEW_CERTIFICATES, course):
-            if course.certificate_available_date and not course.self_paced:
+            if (
```

Comment:
I believe you need to take END into account here still.

I am wondering if we should eventually add a ticket to point all of these checks to one method eventually. I am not advocating for it in this PR, you already have a lot to deal with here.

Reply:
Like you did here: https://github.com/edx/edx-platform/pull/27830/files#diff-edb3995ebb3452c122e32954149f176d38476d0c6b960164d09ff48f8d9a5f51R515-R520

Maybe something like `certificates_viewable_for_course` in the future

Reply:
This will edit the Cert issues date of certs I think. Should we set the modified date to `end_date`? 🤔 I might be overthinking this so this is not a blocking comment.

Reply:
Updated to use `course.end` if set to `END`. I think this fixes the concern.